portfolio/src/main/webapp/script.js

```diff
@@ -162,24 +162,36 @@ function topFunction() {
   document.documentElement.scrollTop = 0;
 }
 
-async function loadComments() {
-  fetch('/comments-list').then(response => response.json()).then((comments) => {
+function loadComments() {
+  document.getElementById("commentsList").innerHTML = "";
+  fetch(`/comments-list?max=${getNumberOfComments()}`).then(response => response.json()).then((comments) => {
     const commentListElement = document.getElementById('commentsList');
     comments.forEach((comment) => {
       commentListElement.appendChild(createCommentElement(comment));
     })
   });
 }
 
+function deleteComments() { 
+  fetch("/delete-comments", {method: "Post"}).then(response => {loadComments()}); 
+}
+
 function createCommentElement(comment) {
   const commentElement = document.createElement('li');
   commentElement.className = 'comment';
 
   const contentElement = document.createElement('span');
   contentElement.innerText = comment.content;
+  const dateElement = document.createElement('p');
+  dateElement.innerText = String(comment.currentDate);
 
   commentElement.appendChild(contentElement);
+  commentElement.appendChild(dateElement);
   return commentElement;
 }
 
-
+function getNumberOfComments() {
+  const numberOfCommentsMenu = document.getElementById("numberOfComments");
+  let numberOfComments = numberOfCommentsMenu.options[numberOfCommentsMenu.selectedIndex].value;
```

Comment:
Optional Nit: You could just return this value directly. The function is called "getNumberOfComments", so the meaning of the expression should be clear.

```
return numberOfCommentsMenu.options[numberOfCommentsMenu.selectedIndex].value;
```